### ISSUE TITLE
Think about production dependencies

### ISSUE THREAD

**Issue by leipert** [digitalbazaar/jsonld.js#152]
Hey there.

I was just reading the code and wondered about a few dependencies/possible improvements:
- `pkginfo`: could be removed if the source code of [L8133-L8135](https://github.com/digitalbazaar/jsonld.js/blob/master/js/jsonld.js#L8133-L8135) is changed to:
  
  ```
  jsonld.version = require('../package.json').version
  ```

And this is somewhat related to #99:
- `request` is used for the node document loader. would it make sense to split json.ld into two modules, so that the json ld processor becomes a separate module, which is somewhat smaller and does not include the document loaders
- `xmldom` same as above, only used for rdfa converting, maybe provide it as a plugin

If you are somewhat happy about talking those improvements, I would be happy to help you with changes.

**Comment by leipert** [comment]
I just found #126 and #136.

**Comment by svanzoest** [comment]
I am currently running info an issue with webpack not dealing well with `pkginfo` for `jsonld`

```
test.js:110784
	    throw new Error('Could not find package.json up from ' +
	    ^

Error: Could not find package.json up from 758
    at Function.pkginfo.find (test.js:110784:12)
    at Function.pkginfo.read (test.js:110811:23)
    at module.exports (test.js:110757:22)
    at Object.<anonymous> (test.js:110818:2)
    at Object.<anonymous> (test.js:110824:31)
    at __webpack_require__ (test.js:21:30)
    at wrapper (test.js:109286:4)
    at test.js:109312:4
    at Object.<anonymous> (test.js:109332:4)
    at Object.module.exports.hasExcape (test.js:109334:31)
```

So the work on #126 is appreciated, especially to make it portable to other environments.

**Comment by davidlehn** [comment]
pkginfo usage was removed in https://github.com/digitalbazaar/jsonld.js/commit/17ec94a97a11576d319db643829f720db821d8a1 and released in 0.5.17.

**Comment by dlongley** [comment]
Closing as fixed or now duplicates #399 in some way.
